cargo-fix/src/main.rs

```diff
@@ -228,6 +244,30 @@ fn rustfix_crate(rustc: &Path, filename: &str) -> Result<HashMap<String, String>
     Ok(old_files)
 }
 
+fn log_for_human(kind: &str, msg: &str) -> Result<(), Error> {
```

Comment:
Hm so thinking about this I'm not actually sure that this'll be the best place to execute this. We've got a bunch of concurrent invocations of the rustc shim running and we don't necessarily want the output to get all jumbled up. The compiler fixed this by line-buffering on Unix and using a global lock on Windows.

I think that this implementation shoudl be fine on Unix due to line buffering (no jumbling), but I think that this implementation runs the risk of jumbling things up on Windows with concurrent invocations.

Two possible ways to fix this could be:

* Send messages through a TCP socket to the main `cargo fix` invocation to globally serialize them there
* Use a global mutex on Windows

I'd be slightly tempted to do the previous one, what do you think?

Reply:
Oh, wow, I didn't even know Windows had no line buffering!

Yeah, let's try it with a TCP socket. This should keep the code consistent across platforms, too. I'd probably try and pass around just the data, and make it human readable string in the main server.

Want to add that here or should be make a follow up PR?

Reply:
I'd prefer to make sure this is implemented before landing but I can take a look into implementing it if you're busy!

Reply:
Alright, I'll whip something up!